nilearn/glm/tests/test_first_level.py

```diff
@@ -1860,3 +1861,86 @@ def test_missing_trial_type_column_warning(tmp_path_factory):
             "No column named 'trial_type' found" in r.message.args[0]
             for r in record
         )
+
+
+# -----------------------surface tests--------------------------------------- #
+
+
+@pytest.fixture()
+def _make_surface_glm_data(rng, make_mini_img):
+    """Create a surface image and design matrix for testing."""
+
+    def _make_surface_image(shape=5):
+        des = pd.DataFrame(
+            rng.standard_normal((shape, 3)), columns=["", "", ""]
+        )
+        mini_img = make_mini_img((shape,))
+        return mini_img, des
+
+    return _make_surface_image
+
+
+def test_flm_fit_surface_image(_make_surface_glm_data):
+    """Test FirstLevelModel with surface image and mask_img set to False."""
+    mini_img, des = _make_surface_glm_data(5)
```

Comment:
it takes two arguments, no ?

Reply:
It works a bit differently within the context of pytest which will automatically pass the 2 arguments (which are actually other fixtures) to the _make_surface_glm_data fixture which then calls the inner function that takes one argument of shape that I'm passing here